File: izinga-ordermanager/src/main/java/io/curiousoft/izinga/ordermanagement/orders/OrderService.java
package io.curiousoft.izinga.ordermanagement.orders;

import io.curiousoft.izinga.commons.model.Order;
import org.springframework.scheduling.annotation.Scheduled;

import java.util.List;

public interface OrderService {

    Order startOrder(Order order) throws Exception;

    Order finishOder(Order order) throws Exception;

    Order findOrder(String orderId);

    Order progressNextStage(String orderId) throws Exception;

    Order applyPromoCode(String promoCode, Order order) throws Exception;

    List<Order> findOrderByUserId(String userId);

    List<Order> findOrderByPhone(String userId) throws Exception;

    List<Order> findOrderByStoreId(String shopId) throws Exception;

    List<Order> findOrderByMessengerId(String id);

    List<Order> findAll();

    Order cancelOrder(String id) throws Exception;
}

File: izinga-commons/src/main/kotlin/io/curiousoft/izinga/commons/order/OrderService.kt
package io.curiousoft.izinga.commons.order

import io.curiousoft.izinga.commons.model.Order

interface OrderService {
    fun startOrder(order: Order?): Order?

    fun finishOder(order: Order?): Order?

    fun findOrder(orderId: String?): Order?

    fun progressNextStage(orderId: String?): Order?

    fun applyPromoCode(promoCode: String?, order: Order?): Order?

    fun findOrderByUserId(userId: String?): MutableList<Order?>?

    fun findOrderByPhone(userId: String?): MutableList<Order?>?

    fun findOrderByStoreId(shopId: String?): MutableList<Order?>?

    fun findOrderByMessengerId(id: String?): MutableList<Order?>?

    fun findAll(): MutableList<Order?>?

    fun cancelOrder(id: String?): Order?
}
